src/sage/topology/simplicial_complex.py

```diff
@@ -4950,6 +4950,61 @@ def bigraded_betti_number(self, a, b, base_ring=ZZ):
 
         return B
 
+    def is_golod(self):
+        r"""
+        Return whether ``self`` is Golod.
+
+        A simplicial complex is Golod if multiplication and all higher
+        Massey operations in the associated Tor-algebra are trivial. This
+        is done by checking the bigraded Betti numbers.
+
+        EXAMPLES::
+
+            sage: X = SimplicialComplex([[0,1],[1,2],[2,3],[3,0]])
+            sage: Y = SimplicialComplex([[0,1,2],[0,2],[0,4]])
+            sage: X.is_golod()
+            False
+            sage: Y.is_golod()
+            True
+        """
+        H = set(a+b for (a, b) in self.bigraded_betti_numbers())
+        if 0 in H:
+            H.remove(0)
+
+        return not any(i + j in H for i in H for j in H)
```

Comment:
I think I changed the logic slightly for `is_golod`. Before, it was possible that `i + j == 0`, which would have returned `False`. However, the current code doesn't catch this. Also, with the commutativity, we can simplify it.
```suggestion
        H = set(a+b for (a, b) in self.bigraded_betti_numbers())
        if 0 in H:
            HL = list(H.difference([0]))
        else:
            HL = list(H)

        return not any(i + j in H for ii, i in enumerate(HL) for j in HL[ii:])
```